### PR TITLE
Holocene: extend op-node to support configurable eip-1559 params

### PR DESCRIPTION
**Pull request by roberto-bayardo**
**Description**

Add support for providing EIP-1559 params to the execution engine as required by Holocene upgrade.

**Tests**

Unit test added to make sure eip1559 parameters are specified properly in the payload attributes post holocene.

e2e tests will be in a followup PR.

**Additional context**

**Metadata**

### CODE REVIEW

File: packages/contracts-bedrock/src/L1/SystemConfig.sol [hunk 1]
```diff
@@ -23,7 +23,7 @@ import { IResourceMetering } from "src/L1/interfaces/IResourceMetering.sol";
 contract SystemConfig is OwnableUpgradeable, ISemver, IGasToken {
     /// @notice Enum representing different types of updates.
     /// @custom:value BATCHER              Represents an update to the batcher hash.
-    /// @custom:value GAS_CONFIG           Represents an update to txn fee config on L2.
+    /// @custom:value FEE_SCALARS          Represents an update to l1 data fee scalars.
```
Comment (tynes) [inline packages/contracts-bedrock/src/L1/SystemConfig.sol:26]: Can you also rename it in the `UpdateType`? There may be places in `op-node` that need the name change as well

**Comment by tynes** [comment]
Not sure why the interop tests are failing but you can skip them for now

**Comment by tynes** [comment]
I would like to remove as much boilerplate as possible when it comes to the spots that all need to be updated to add a new hardfork

File: op-node/rollup/derive/system_config.go [hunk 2]
```diff
@@ -140,6 +141,22 @@ func ProcessSystemConfigUpdateLogEvent(destSysCfg *eth.SystemConfig, ev *types.L
 		}
 		destSysCfg.GasLimit = gasLimit
```
Comment (tynes) [inline op-node/rollup/derive/system_config.go:142]: One thought is do we want to gate this behind an `isHolocene` flag?

Some options:
- If not is holocene, error on unknown event type
- If not is holocene, recognize the event type but do not parse it
- Always parse it, the L1 attributes tx will just not utilize the parsed data

I do not like erroring on unknown event type, it adds complexity with contract versions and op-node versions. I am supportive of the current implementation, just raising this for thoughts

File: packages/contracts-bedrock/src/L1/SystemConfig.sol (hunk 1 above)
Comment (roberto-bayardo) [reply packages/contracts-bedrock/src/L1/SystemConfig.sol:26]: yup will do

Comment (roberto-bayardo) [reply packages/contracts-bedrock/src/L1/SystemConfig.sol:26]: done

File: op-node/rollup/derive/system_config.go (hunk 2 above)
Comment (roberto-bayardo) [reply op-node/rollup/derive/system_config.go:142]: Doesn't seem there's much harm in allowing it to be parsed. It just means as soon as the holocene upgrade takes place, the new values will take effect, which is actually kind of convenient? (No need to wait until after the fork to set up your desired parameters....)

**Comment by roberto-bayardo** [comment]
> I would like to remove as much boilerplate as possible when it comes to the spots that all need to be updated to add a new hardfork

This PR should be a good reference for whoever takes this task on.

**Comment by roberto-bayardo** [comment]
Do not merge yet: currently depending on my op-geth fork ( https://github.com/ethereum-optimism/op-geth/pull/402 )

**Comment by threewebcode** [comment]
The code change covers two aspects: 1. add holocene time; 2. add EIP-1559 parameters into execution payload.

What are the benefits of adding EIP-1559 parameters into execution payload?

**Comment by roberto-bayardo** [comment]
replacing with https://github.com/ethereum-optimism/optimism/pull/12496

**Pull request closed without merging**